src/main/kotlin/dev/arbjerg/lavalink/client/loadbalancing/builtin/INodeHealthProvider.kt

```diff
@@ -0,0 +1,27 @@
+package dev.arbjerg.lavalink.client.loadbalancing.builtin
+
+import dev.arbjerg.lavalink.client.loadbalancing.MAX_ERROR
+import dev.arbjerg.lavalink.protocol.v4.Message
+
+interface INodeHealthProvider {
```

Comment:
Would it be a good idea to turn this into an abstract class? That way you could enforce a default constructor that always takes the lavalink node as argument. I feel like it would be difficult to get at some of the required stats from the node otherwise. I feel like builder can be changed to accept a function with the node as single argument for the provider